components/motor/gpio/motor_encoder.go

```diff
@@ -446,7 +446,16 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 		return err
 	}
 
-	return m.opMgr.WaitTillNotPowered(ctx, time.Millisecond, m)
+	if err := m.opMgr.WaitTillNotPowered(ctx, time.Millisecond, m); err != nil {
+		if errStop := m.Stop(ctx, map[string]interface{}{}); errStop != nil {
+			m.logger.Error("Error stopping the motor")
```

Comment:
think we need to log or just return?

Reply:
I think if gofor is used in a goroutine having the logging message would be helpful for users to figure out what is happening. But im fine with either.